Materials2050_Adapter/CRUD/Read/Read.cs

```diff
@@ -0,0 +1,181 @@
+                            apiName = "get_products";
+                            BH.Engine.Base.Compute.RecordError($"The {apiT} API has not yet been implemented. Please use the OpenAPI.");
+                            break;
+                        }
+                    case APIName.GlobalWarmingAPI:
+                        {
+                            apiName = "get_co2_warming_potential";
+                            BH.Engine.Base.Compute.RecordError($"The {apiT} API has not yet been implemented. Please use the OpenAPI.");
+                            break;
+                        }
+                    default:
+                        apiName = "";
+                        BH.Engine.Base.Compute.RecordError("Please select which API you would like to query.");
+                        break;
+                }
+            }
+            else
+            {
+                BH.Engine.Base.Compute.RecordWarning("No config was provided. Using OpenAPI as the default 2050 Materials API for all queries.");
+                apiName = "get_products_open_api";
+            }
+
+            // Refresh the API Token automatically
+            string refreshedToken = BH.Engine.Adapters.Materials2050.Create.RefreshAPIToken(m_apiToken);
+
+            //Create GET Request
+            GetRequest epdGetRequest = BH.Engine.Adapters.Materials2050.Create.Materials2050Request(apiName, refreshedToken, config);
+
+            string response = BH.Engine.Adapters.HTTP.Compute.MakeRequest(epdGetRequest);
+            List<object> responseObjs = new List<object>();
+
+            if (string.IsNullOrEmpty(response))
+            {
+                BH.Engine.Base.Compute.RecordWarning("No response received, API token and connection.");
+                return null;
+            }
+            //Check if the response is a valid json
+            else if (response.StartsWith("{"))
+            {
+                response = "[" + response + "]";
+                responseObjs = new List<object>() { Engine.Serialiser.Convert.FromJsonArray(response) };
+            }
+            else if (response.StartsWith("["))
+            {
+                responseObjs = new List<object>() { Engine.Serialiser.Convert.FromJsonArray(response) };
+            }
+            else
+            {
+                BH.Engine.Base.Compute.RecordWarning("Response is not a valid JSON. Please check your config and try again.");
+                return null;
+            }
+
+            //Convert nested customObject from serialization to list of epdData objects
+            List<EnvironmentalProductDeclaration> epds = new List<EnvironmentalProductDeclaration>();
+            EnvironmentalProductDeclaration epd = new EnvironmentalProductDeclaration();
+            
+            if(responseObjs.Count >= 0) 
+            {
+                object epdObjects = responseObjs[0]; // we always want the zero index item. In the event it's the wrong data, other errors will be triggered.
```

Comment:
```suggestion
            if(responseObjs.Count > 0) 
            {
                object epdObjects = responseObjs[0]; // we always want the zero index item. In the event it's the wrong data, other errors will be triggered.
```

If the `Count` is `== 0` then you won't be able to access the `0` index on the next line, throwing an Index Out of Range exception